quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -617,73 +455,137 @@ impl Metastore for PostgresqlMetastore {
     async fn publish_splits<'a>(
         &self,
         index_id: &str,
-        new_split_ids: &[&'a str],
+        staged_split_ids: &[&'a str],
         replaced_split_ids: &[&'a str],
         checkpoint_delta_opt: Option<IndexCheckpointDelta>,
     ) -> MetastoreResult<()> {
         run_with_tx!(self.connection_pool, tx, {
+            let mut index_metadata = index_metadata(tx, index_id).await?;
```

Comment:
We can remove this transaction when we move index checkpoints to a dedicated table. This is already doable, but the JSON query is awful.